namesys/publisher.go

```diff
@@ -105,6 +106,11 @@ func ipnsEntryDataForSig(e *pb.IpnsEntry) []byte {
 		[]byte{})
 }
 
+var IpnsRecordValidator = &record.ValidChecker{
+	Func: ValidateIpnsRecord,
+	Sign: true,
```

Comment:
If we remove the purity from the validators then i can make them just a simple function